metrics/dropwizard-metrics/src/main/java/org/apache/iotdb/metrics/dropwizard/reporter/DropwizardPrometheusReporter.java

```diff
@@ -20,66 +20,89 @@
 package org.apache.iotdb.metrics.dropwizard.reporter;
 
 import org.apache.iotdb.metrics.MetricManager;
-import org.apache.iotdb.metrics.Reporter;
 import org.apache.iotdb.metrics.config.MetricConfig;
 import org.apache.iotdb.metrics.config.MetricConfigDescriptor;
 import org.apache.iotdb.metrics.dropwizard.DropwizardMetricManager;
-import org.apache.iotdb.metrics.dropwizard.reporter.prometheus.PrometheusReporter;
-import org.apache.iotdb.metrics.dropwizard.reporter.prometheus.PushGateway;
+import org.apache.iotdb.metrics.reporter.Reporter;
 import org.apache.iotdb.metrics.utils.ReporterType;
 
-import com.codahale.metrics.MetricFilter;
+import com.codahale.metrics.MetricRegistry;
+import io.netty.channel.ChannelOption;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
+import reactor.core.publisher.Mono;
+import reactor.netty.DisposableServer;
+import reactor.netty.http.server.HttpServer;
 
-import java.util.concurrent.TimeUnit;
+import java.io.IOException;
+import java.io.StringWriter;
+import java.io.Writer;
+import java.time.Duration;
 
 public class DropwizardPrometheusReporter implements Reporter {
   private static final Logger LOGGER = LoggerFactory.getLogger(DropwizardPrometheusReporter.class);
   private static final MetricConfig metricConfig =
       MetricConfigDescriptor.getInstance().getMetricConfig();
 
   private MetricManager dropwizardMetricManager = null;
-  private PrometheusReporter prometheusReporter = null;
+  private DisposableServer httpServer = null;
 
   @Override
   public boolean start() {
-    if (prometheusReporter != null) {
+    if (httpServer != null) {
       LOGGER.warn("Dropwizard Prometheus Reporter already start!");
       return false;
     }
-    String url = metricConfig.getPrometheusReporterConfig().getPrometheusExporterUrl();
-    String port = metricConfig.getPrometheusReporterConfig().getPrometheusExporterPort();
-    PushGateway pushgateway = new PushGateway(url + ":" + port, "IoTDB-Metric{Dropwizard}");
+    httpServer =
+        HttpServer.create()
+            .idleTimeout(Duration.ofMillis(30_000L))
+            .option(ChannelOption.CONNECT_TIMEOUT_MILLIS, 2000)
+            .port(Integer.parseInt(metricConfig.getPrometheusExporterPort()))
+            .route(
+                routes ->
+                    routes.get(
+                        "/metrics",
+                        (request, response) -> response.sendString(Mono.just(scrape()))))
+            .bindNow();
+
+    LOGGER.info(
+        "http server for metrics started, listen on {}", metricConfig.getPrometheusExporterPort());
+    return true;
+  }
+
+  private String scrape() {
+    MetricRegistry metricRegistry =
+        ((DropwizardMetricManager) dropwizardMetricManager).getMetricRegistry();
+    Writer writer = new StringWriter();
```

Comment:
Is the writer auto closable?

Reply:
StringWriter isn't need to close.
![image](https://user-images.githubusercontent.com/46039728/156494953-087d9066-c3bc-485a-af8e-0a12b91d29f6.png)